python/cugraph/structure/graph.py

```diff
@@ -565,945 +277,394 @@ def from_numpy_matrix(self, np_matrix):
-        df["out_degree"] = out_degree_col
-
-        if self.renumbered is True:
-            df = self.unrenumber(df, "vertex")
-
-        if vertex_subset is not None:
-            df = df[df['vertex'].isin(vertex_subset)]
-
-        return df
-
-    def _degree(self, vertex_subset, x=0):
-        vertex_col, degree_col = graph_primtypes_wrapper._degree(self, x)
-        df = cudf.DataFrame()
-        df["vertex"] = vertex_col
-        df["degree"] = degree_col
-
-        if self.renumbered is True:
-            df = self.unrenumber(df, "vertex")
-
-        if vertex_subset is not None:
-            df = df[df['vertex'].isin(vertex_subset)]
-
-        return df
+        return self.properties.isolated_vertices
 
     def to_directed(self):
         """
         Return a directed representation of the graph.
         This function sets the type of graph as DiGraph() and returns the
         directed view.
-
         Returns
         -------
         G : DiGraph
             A directed graph with the same nodes, and each edge (u,v,weights)
             replaced by two directed edges (u,v,weights) and (v,u,weights).
-
         Examples
         --------
         >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
         >>>                   dtype=['int32', 'int32', 'float32'], header=None)
         >>> G = cugraph.Graph()
         >>> G.from_cudf_edgelist(M, '0', '1')
         >>> DiG = G.to_directed()
-
         """
-        if self.distributed:
-            raise Exception("Not supported for distributed graph")
-        if type(self) is DiGraph:
+        if self.graph_properties.directed is True:
             return self
-        if type(self) is Graph:
-            DiG = DiGraph()
-            DiG.renumbered = self.renumbered
-            DiG.renumber_map = self.renumber_map
-            DiG.edgelist = self.edgelist
-            DiG.adjlist = self.adjlist
-            DiG.transposedadjlist = self.transposedadjlist
-            return DiG
+        else:
+            directed_graph = type(self)(directed=True)
+            directed_graph.Impl = type(self.Impl)(directed_graph.
+                                                  graph_properties)
+            self.Impl.to_directed(directed_graph.Impl)
+            return directed_graph
 
     def to_undirected(self):
         """
         Return an undirected copy of the graph.
-
         Returns
         -------
         G : Graph
             A undirected graph with the same nodes, and each directed edge
             (u,v,weights) replaced by an undirected edge (u,v,weights).
-
         Examples
         --------
         >>> M = cudf.read_csv('datasets/karate.csv', delimiter=' ',
         >>>                   dtype=['int32', 'int32', 'float32'], header=None)
         >>> DiG = cugraph.DiGraph()
         >>> DiG.from_cudf_edgelist(M, '0', '1')
         >>> G = DiG.to_undirected()
-
         """
 
-        if type(self) is Graph:
+        if self.graph_properties.directed is False:
```

Comment:
maybe use `hasattr` for consistency?

Reply:
Graph will always have the directed attribute. All Graphs will have this property set during initialization itself. Other properties need to be set only in specific graphs